modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/search/engine/adapter/search/SearchSearchRequestExecutorImpl.java

```diff
@@ -111,6 +125,31 @@ private String _getPrettyPrintedRequestString(
 		}
 	}
 
+	private SearchResponse _getScrollSearchResponse(
+		SearchSearchRequest searchSearchRequest) {
+
+		RestHighLevelClient restHighLevelClient =
+			_elasticsearchClientResolver.getRestHighLevelClient(
+				searchSearchRequest.getConnectionId(),
+				searchSearchRequest.isPreferLocalCluster());
+
+		SearchScrollRequest searchScrollRequest = new SearchScrollRequest(
+			searchSearchRequest.getScrollId());
+
+		searchScrollRequest.scroll(
```

Comment:
maybe the null check for getScrollSearchTimeValue should be done here and if there isnt one set we just let elastic use a default? not sure what happens if you pass null to searchScrollRequest.scroll?

Reply:
also, i kind of like the name "scrollKeepAlive" instead of "scrollSearchTimeValue". maybe you can change that as well